onnx/optimizer/optimize.h

```diff
@@ -43,32 +44,31 @@ struct Optimizer {
       return mp_in;
     }
 
+    onnx::ModelProto mp_out{};
+    PrepareOutput(mp_in, mp_out);
+
     for (auto & name : names) {
       auto it = passes.find(name);
       ONNX_ASSERTM(it != passes.end(), "pass %s is unknown.", name.c_str());
       if (it != passes.end()) {
         auto& pass = it->second;
         if (pass->type == API_TYPE::PROTO) {
           // Operate on ModelProto.
-          std::unique_ptr<onnx::ModelProto> temp_out(new ModelProto());
-          PrepareOutput(*mp_in, *temp_out);
-          ExportModelProto(temp_out.get(), g);
-          pass->optimize(*temp_out);
-          g = onnx::ImportModelProto(*temp_out);
-          mp_in = std::move(temp_out);
+          ExportModelProto(&mp_out, g);
+          pass->optimize(mp_out);
+          g = onnx::ImportModelProto(mp_out);
         } else {
           // Operate on Graph (IR).
           pass->optimize(*g);
         }
       }
     }
-    std::unique_ptr<onnx::ModelProto> mp_out(new ModelProto());
-    PrepareOutput(*mp_in, *mp_out);
-    ExportModelProto(mp_out.get(), g);
+    ExportModelProto(&mp_out, g);
     return mp_out;
```

Comment:
For the new implementation, we replies on RVO. Otherwise, we will have additional copy operation.